crates/uv-toolchain/src/discovery.rs

```diff
@@ -546,6 +546,12 @@ pub fn find_toolchains<'a>(
                 debug!("Checking for Python interpreter at {request}");
                 Box::new(
                     python_interpreters_with_executable_name(name, cache)
+                        .filter(move |result| {
+                            result.is_err()
+                                || result.as_ref().is_ok_and(|(source, interpreter)| {
+                                    satisfies_system_python(*source, interpreter, system)
+                                })
```

Comment:
Hmmm maybe, `result.as_ref().ok().map_or(true, |(source, interpreter)| { ... })`?

I still might slap this into a helper with a nice name though if you need to repeat it a lot.